### PR TITLE
Fixed preview link to use parent resource link if child has none

### LINKED ISSUE #814: Bad preview links in production
Reported by Chen in https://odl.zendesk.com/agent/tickets/2557

> The "Missing Title" issue might be more complicated than that. Try searching for "angular momentum homework" in 8.MReVx, and looked for problems. On the first page you'll see something like this:
> 
> If you click preview on the "Missing Title" ones, they'll actually bring you to a blank course page. The vertical "checkpoint 2: Momentum of a System" only contains two problems, both have names and are found. The two "Missing Title" ones are actually ghost components that don't exist (or at least invisible to me).

In my investigation, it appears that some of the (problem) learning resources have bad preview links that are missing the `/jump_to_id/` followed by `url_name`. 

For example, from this search:

http://lore.odl.mit.edu/repositories/classical-mechanics/?selected_facets=course_exact%3A8.MReVx&selected_facets=resource_type_exact%3Aproblem

these resources have bad preview URLs:

http://lore.odl.mit.edu/api/v1/repositories/classical-mechanics/learning_resources/21192/
http://lore.odl.mit.edu/api/v1/repositories/classical-mechanics/learning_resources/21271/

I can't imagine why, but it seems that every resource with a missing title has a bad preview URL.

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #814

### CODE REVIEW

File: learningresources/models.py [hunk 1]
```diff
@@ -216,12 +216,25 @@ def get_preview_url(resource, org=None, course_number=None, run=None):
         run=run,
     )
 
-    if resource.url_name is None:
+    url_name = None
```
Comment (ShawnMilo) [inline learningresources/models.py]: nit:

Lines 219 - 221 can just be `url_name = resource.url_name`, and line 222 can be `if url_name is None`. The `else` makes it _slightly_ more difficult to follow.

File: importer/tests/test_import.py [hunk 2]
```diff
@@ -291,3 +299,46 @@ def get_counts():
...
+        self.assertEqual(html_resources.count(), 1)
+        html_resource = html_resources.first()
+        self.assertEqual(
+            get_preview_url(html_resource),
+            "{base}courses/{org}/{course}/{run}/jump_to_id/{url_path}".format(
+                base=settings.LORE_PREVIEW_BASE_URL,
+                org=html_resource.course.org,
+                course=html_resource.course.course_number,
+                run=html_resource.course.run,
+                url_path="2015_Summer"
```
Comment (ShawnMilo) [inline importer/tests/test_import.py:364]: This test is good. Do you think having a test for the case of no title anywhere in the tree would be beneficial?

Comment (noisecapella) [inline importer/tests/test_import.py:364]: It's a good idea, I'll add it

**Comment by noisecapella** [comment]
Your comments should be addressed now, anything else?

**Comment by ShawnMilo** [comment]
No, nothing else. 

**Pull request merged**